libs/utils/analysis/thermal_analysis.py

```diff
@@ -0,0 +1,243 @@
+
+    def _dfg_thermal_zone_temperature(self, ids=None):
+        """
+        Get the temperature readings of one or more thermal zone(s).
+
+        :param ids: The thermal zones to consider
+        :type ids: list(int)
+        """
+        df = self._dfg_trace_event('thermal_temperature')
+        df = df[['id', 'thermal_zone', 'temp']]
+
+        if ids is not None:
+            df = df[df.id.isin(ids)]
+
+        return df
+
+    def _dfg_cpufreq_cooling_state(self, cpus=None):
+        """
+        Get the cpufreq cooling device states
+
+        :param cpus: The cpus to consider
+        :type cpus: list(int)
+        """
+        df = self._dfg_trace_event('thermal_power_cpu_limit')
+        df = df[['cpus', 'freq', 'cdev_state']]
+
+        if cpus is not None:
+            # Find masks that match the requested CPUs
+            # This can include other CPUs
+            masks = self._matchingMasks(cpus)
+            df = df[df.cpus.isin(masks)]
+
+        return df
+
+    def _dfg_devfreq_cooling_state(self, devices=None):
+        """
+        Get the temperature readings of one or more thermal zone(s).
+
+        :param devices: The devfreq devices to consider
+        :type device: list(str)
+        """
+        df = self._dfg_trace_event('thermal_power_devfreq_limit')
+        df = df[['type', 'freq', 'cdev_state']]
+
+        if devices is not None:
+            df = df[df.type.isin(devices)]
+
+        return df
+
+
+###############################################################################
+# Plotting Methods
+###############################################################################
+
+    def plotTemperature(self, thermal_zones=None, ax=None):
+        """
+        Plot temperature of thermal zones (all by default)
+
+        Requires the following trace event:
+            - thermal_temperature
+
+        :param thermal_zones: ID(s) of the zones to be plotted.
+            All the zones are plotted by default.
+            IDs can be found in syfs:  /sys/class/thermal/thermal_zone<ID>
+        :type thermal_zones: list(int)
+        """
+        if not self._trace.hasEvents('thermal_temperature'):
+            self._log.warning('Event [{}] not found, plot DISABLED!'.format(event))
+            return
+
+        plot_df = self._dfg_thermal_zone_temperature(thermal_zones)
+
+        def stringify_tz(id):
+            return plot_df[plot_df.id == id]['thermal_zone'].unique()[0]
+
+        plot_generic(plot_df, 'id', names = thermal_zones,
+                     fields = ['temp'], prettify_name = stringify_tz,
+                     xlim = (self._trace.x_min, self._trace.x_max),
+                     drawstyle = 'steps-post', ax = ax
+        )
+
+        if thermal_zones is None:
+            suffix = ''
+        else:
+            suffix = '_' + '_'.join(map(str, thermal_zones))
+
+        # Save generated plots into datadir
+        figname = '{}/{}thermal_temperature{}.png'\
```

Comment:
`os.path.join`. Also you can avoid the `\` by putting the whole thing in brackets like

```
   foo = ('bar{}'
          .format('baz'))

Reply:
Do you want me to do a pass around the other plotting methods ? I copied the boilerplate code from frequency_analysis but it looks like it's done this way in the other plotting methods as well.

Reply:
Oh OK, it doesn't matter too much really.